### PR TITLE
Add fileshare controller unit test cases for create/delete fileshare and create/delete filesharesnapshots

### PR DESCRIPTION
**Pull request by Shruthi-1MN**
**What this PR does / why we need it**:
Add fileshare controller unit test cases for create/delete fileshare and create/delete filesharesnapshots

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/controller/fileshare/filesharecontroller_test.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+	}
+}
+
+func TestCreateFileShare(t *testing.T) {
+	fc := NewFakeController()
+	var expected = &SampleShares[0]
+
+	result, err := fc.CreateFileShare(&pb.CreateFileShareOpts{})
+	if err != nil {
+		t.Errorf("Failed to create fileshare, err is %v\n", err)
```
Comment (leonwanghui) [inline pkg/controller/fileshare/filesharecontroller_test.go]: The first letter of error msg should be lower case

File: pkg/controller/fileshare/filesharecontroller_test.go [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+	}
+
+	if !reflect.DeepEqual(result, expected) {
+		t.Errorf("Expected %v, got %v\n", expected, result)
```
Comment (leonwanghui) [inline pkg/controller/fileshare/filesharecontroller_test.go]: The first letter of error msg should be lower case

File: pkg/controller/fileshare/filesharecontroller_test.go [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+	}
+}
+
+func TestDeleteFileShare(t *testing.T) {
+	fc := NewFakeController()
+
+	result := fc.DeleteFileShare(&pb.DeleteFileShareOpts{})
+	if result != nil {
+		t.Errorf("Expected %v, got %v\n", nil, result)
```
Comment (leonwanghui) [inline pkg/controller/fileshare/filesharecontroller_test.go]: The first letter of error msg should be lower case

File: pkg/controller/fileshare/filesharecontroller_test.go [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+	}
+}
+
+func TestCreateFileShareSnapshot(t *testing.T) {
+	fc := NewFakeController()
+	var expected = &SampleShareSnapshots[0]
+
+	result, err := fc.CreateFileShareSnapshot(&pb.CreateFileShareSnapshotOpts{})
+	if err != nil {
+		t.Errorf("Failed to create file share snapshot, err is %v\n", err)
```
Comment (leonwanghui) [inline pkg/controller/fileshare/filesharecontroller_test.go]: The first letter of error msg should be lower case

File: pkg/controller/fileshare/filesharecontroller_test.go [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+	}
+
+	if !reflect.DeepEqual(result, expected) {
+		t.Errorf("Expected %v, got %v\n", expected, result)
```
Comment (leonwanghui) [inline pkg/controller/fileshare/filesharecontroller_test.go]: The first letter of error msg should be lower case

File: pkg/controller/fileshare/filesharecontroller_test.go [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+	}
+}
+
+func TestDeleteFileShareSnapshot(t *testing.T) {
+	fc := NewFakeController()
+
+	result := fc.DeleteFileShareSnapshot(&pb.DeleteFileShareSnapshotOpts{})
+	if result != nil {
+		t.Errorf("Expected %v, got %v\n", nil, result)
```
Comment (leonwanghui) [inline pkg/controller/fileshare/filesharecontroller_test.go]: The first letter of error msg should be lower case

File: pkg/controller/fileshare/filesharecontroller_test.go (hunk 1 above)
Comment (Shruthi-1MN) [reply pkg/controller/fileshare/filesharecontroller_test.go]: Done

File: pkg/controller/fileshare/filesharecontroller_test.go (hunk 2 above)
Comment (Shruthi-1MN) [reply pkg/controller/fileshare/filesharecontroller_test.go]: Done

File: pkg/controller/fileshare/filesharecontroller_test.go (hunk 3 above)
Comment (Shruthi-1MN) [reply pkg/controller/fileshare/filesharecontroller_test.go]: Done

File: pkg/controller/fileshare/filesharecontroller_test.go (hunk 5 above)
Comment (Shruthi-1MN) [reply pkg/controller/fileshare/filesharecontroller_test.go]: Done

File: pkg/controller/fileshare/filesharecontroller_test.go (hunk 6 above)
Comment (Shruthi-1MN) [reply pkg/controller/fileshare/filesharecontroller_test.go]: Done

File: pkg/controller/fileshare/filesharecontroller_test.go (hunk 4 above)
Comment (Shruthi-1MN) [reply pkg/controller/fileshare/filesharecontroller_test.go]: Done

**Pull request merged**